kythe/java/com/google/devtools/kythe/platform/java/JavacOptionsUtils.java

```diff
@@ -362,6 +366,56 @@ private ImmutableList<String> removeArgumentPaths(Option option) {
       return paths.build();
     }
 
+    /**
+     * Find any -source flags that specify a version of java that the JRE can't support and replace
+     * them with the lowest version that the JRE does support.
+     *
+     * <p>There may be problems during analysis due to flags passed to javac or language features
+     * that changed, but the alternative is to have the analysis fail immediately when we send the
+     * flag to the underlying Java APIs to do the analysis.
+     */
+    public ModifiableOptions updateToMinimumSupportedSourceVersion() {
+      ArrayList<String> unsupportedVersions = new ArrayList<>();
+      ArrayList<String> supportedVersions = new ArrayList<>();
+      List<String> replacements = new ArrayList<>(internal.size());
+      Consumer<String> matched =
+          (value) -> {
+            Source v = Source.lookup(value);
+            if (v == null) {
+              logger.atWarning().log("Could not parse source version number: %s", value);
+              // Don't mutate the flag if it can't be parsed.
+              supportedVersions.add(value);
+            } else if (v.compareTo(Source.MIN) < 0) {
```

Comment:
`v.isSupported()` also works ;-)

Reply:
That was added sometime between 11 and 19

https://github.com/openjdk/jdk/commit/e18f343141d61ed8382ff6d175aa121e41c95c42